Source/DafnyTestGeneration/README.md

```diff
@@ -0,0 +1,149 @@
+- [ProgramModification.cs](ProgramModification.cs) - represents a modification of the original Boogie program (which is in turn generated in `Main.cs` by translating Dafny input to Boogie). Can be verified and, if verification fails, one can pass the counterexample log as constructor argument to `TestMethod`.
+- [ProgramModifier.cs](ProgramModifier.cs) - an abstract class that contains utilities for annotating Boogie Programs, inlining Dafny methods after they are translated to Boogie, etc. This is all necessary for generating instances of `ProgramModification`
+- [BlockBasedModifier.cs](BlockBasedModifier.cs) and [PathBasedModifier.cs](PathBasedModifier.cs) - subclasses of `ProgramModifier` that generate instances of `ProgramModification` by inserting assertions that fail when a particular block is visited or a particular path is taken, respectively.
+- [DafnyInfo.cs](DafnyInfo.cs) - takes a Dafny Program and extracts certain information about it that is then used in `TestMethod.cs` and `Main.cs`
+
+## How to Generate Tests?
+
+- Test generation currently works with all basic types, user-defined classes, sequences, sets, and maps. It does not work with datatypes, arrays, and multisets. It is also not possible to generate tests for constructors.
+- To generate block- or path-coverage tests use the `/testMode:Block` or `/testMode:Path` arguments respectively. You will likely also need `definiteAssignment:3`, which will ensure that you generate tests even for those methods that do not contain any assertions or pre-\post-conditions.
+- If you wish to test a particular method rather than all the methods in a file, you can specify such a method with the `/testTargetMethod` command line argument and providing the fully qualified method name.
+- If you are using `/testTargetMethod` and would like to inline methods that are called from the method of interest, you can do so by setting `/testInlineDepth` to something larger than zero (zero is the default). The `/verifyAllModules` argument might also be relevant if the methods to be inlined are defined in included files.
+- To deal with loops, you should use `/loopUnroll` and also `/testSeqLengthLimit`. The latter argument adds an axiom that limits the length of any sequence to be no greater than some specified value. This restriction can be used to ensure that the number of loop unrolls is sufficient with respect to the length of any input sequence but it can also cause the program to miss certain corner cases.
+- The`/testMode:DeadCode` argument will make Dafny identify potential dead code in the specified file. Note that false negatives are possible if `/loopUnroll` is not used. False positives are also possible for a variety of reasons, such as `/loopUnroll` being assigned not high enough value.
```

Comment:
I get that generating tests and identifying dead code share a common implementation, but I think they should be exposed as a very different modes of operation. The former produces test files, whereas the latter produces warnings (and I'm hoping we can make it a feature of the language server instead soon).

I would suggest a separate `/warnDeadCode` option, similar to the existing `/warnShadowing`

Reply:
Done! It would indeed be great if one could in the future highlight potentially dead code directly in the IDE using the LanguageServer.